libyul/Dialect.h

```diff
@@ -63,6 +63,9 @@ struct Dialect: boost::noncopyable
 	/// @returns the builtin function of the given name or a nullptr if it is not a builtin function.
 	virtual BuiltinFunction const* builtin(YulString /*_name*/) const { return nullptr; }
 
+	/// @returns true if the identifier is reserved. This includes the builtins too.
```

Comment:
We have at least two options here:
1. (the current implementation) `isReservedIdentifier` contains the superset. This means the creator part needs to include a small amount of code duplication (adding `datasize`, `dataoffset`, etc.)

2. `isReservedIdentifier` excludes the builtins. This makes the creator code more complicated because it has exclude builtins based on EVM version.

Reply:
I think the implementation in the using classes (VarNameCleaner etc) is more important, so the reserved identifiers should include the builtins.

Reply:
I prefer this implementation for simplicity :)